FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheker/HealthCheckEngine.cs

```diff
@@ -0,0 +1,96 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Synapse.HealthCheker.Checkers;
+using Microsoft.Health.Fhir.Synapse.HealthCheker.Models;
+
+namespace Microsoft.Health.Fhir.Synapse.HealthCheker
+{
+    public class HealthCheckEngine : IHealthCheckEngine
+    {
+        private readonly IEnumerable<IHealthChecker> _healthCheckers;
+        private readonly TimeSpan _healthCheckTimeout;
+        private readonly TimeSpan _healthStatusCollectionTimeout;
+        private readonly ILogger<HealthCheckEngine> _logger;
+
+        public HealthCheckEngine(
+            IEnumerable<IHealthChecker> healthCheckers,
+            HealthCheckOptions healthCheckOptions,
+            ILogger<HealthCheckEngine> logger)
+        {
+            EnsureArg.IsNotNull(healthCheckOptions, nameof(healthCheckOptions));
+            _healthCheckers = EnsureArg.IsNotNull(healthCheckers, nameof(healthCheckers));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _healthCheckTimeout = healthCheckOptions.HealthCheckTimeout;
+            _healthStatusCollectionTimeout = healthCheckOptions.HealthStatusCollectionTimeout;
+        }
+
+        public async Task CheckHealthAsync(HealthStatus healthStatus, AsyncCallback callBack = null, CancellationToken cancellationToken = default)
+        {
+            EnsureArg.IsNotNull(healthStatus, nameof(healthStatus));
```

Comment:
I feel this could be a long running health check job with a specific time interval. And for each check, an event could be raised and consumed in the outer service (Or the outer service itsself manages the check logic using a pulling mode).

Reply:
Updated it, thanks.